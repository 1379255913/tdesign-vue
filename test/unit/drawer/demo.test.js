import { mount } from '@vue/test-utils';
import demo from '@/examples/drawer/demos/base.vue';
import CurrentDOMModeUsageExample from '@/examples/drawer/demos/CurrentDOMModeUsageExample.vue';
import destroy from '@/examples/drawer/demos/destroy.vue';
import HeaderUsageExample from '@/examples/drawer/demos/HeaderUsageExample.vue';
import ModeUsageExample from '@/examples/drawer/demos/ModeUsageExample.vue';
import NoMaskUsageExample from '@/examples/drawer/demos/NoMaskUsageExample.vue';
import operation from '@/examples/drawer/demos/operation.vue';
import PlacementUsageExample from '@/examples/drawer/demos/PlacementUsageExample.vue';
import SizeUsageExample from '@/examples/drawer/demos/SizeUsageExample.vue';

// unit test for component in examples.
describe('Drawer', () => {
  it('base demo works fine', () => {
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo CurrentDOMModeUsageExample wroks fine', () => {
    const wrapper = mount(CurrentDOMModeUsageExample);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Demo destroy wroks fine', () => {
    const wrapper = mount(destroy);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo HeaderUsageExample wroks fine', () => {
    const wrapper = mount(HeaderUsageExample);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo ModeUsageExample wroks fine', () => {
    const wrapper = mount(ModeUsageExample);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo NoMaskUsageExample wroks fine', () => {
    const wrapper = mount(NoMaskUsageExample);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo operation wroks fine', () => {
    const wrapper = mount(operation);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo PlacementUsageExample wroks fine', () => {
    const wrapper = mount(PlacementUsageExample);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Demo SizeUsageExample wroks fine', () => {
    const wrapper = mount(SizeUsageExample);
    expect(wrapper.element).toMatchSnapshot();
  });
});
