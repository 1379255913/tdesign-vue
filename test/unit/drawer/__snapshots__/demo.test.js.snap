// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Demo CurrentDOMModeUsageExample wroks fine 1`] = `
<div
  class="t-container"
>
  <div
    class="t-suf-container"
  >
    <div
      class="t-drawer t-drawer-right t-drawer-attach v-transfer-dom"
      style="z-index: 1500;"
    >
      <div
        class="t-drawer__mask"
      />
      <div
        class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
        style="width: 300px;"
      >
        <div
          class="t-drawer__header"
        >
          标题名称
        </div>
        <div
          class="t-drawer__close-btn"
        />
        <div
          class="t-drawer__body"
        >
          <p>
            This is a drawer
          </p>
        </div>
        <div
          class="t-drawer__footer"
        >
          <div>
            <button
              class="t-button t-size-m t-button--variant-outline t-button--theme-default"
              type="button"
            >
              取消
            </button>
            <button
              class="t-button t-size-m t-button--variant-base t-button--theme-default"
              type="button"
            >
              确认
            </button>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="tdesign-radio-button"
    >
      
      抽屉弹出方向：
      
      <div
        class="t-radio-group t-radio-group-outline t-radio-group-medium"
      >
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="left"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            left
          </span>
        </label>
         
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="right"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            right
          </span>
        </label>
         
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="top"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            top
          </span>
        </label>
         
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bottom"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            bottom
          </span>
        </label>
      </div>
    </div>
     
    <br />
     
    <div
      class="tdesign-radio-button"
    >
      
      抽屉弹出模式：
      
      <div
        class="t-radio-group t-radio-group-outline t-radio-group-medium"
      >
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="overlay"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            overlay
          </span>
        </label>
         
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="push"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            push
          </span>
        </label>
      </div>
    </div>
     
    <br />
    <br />
     
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-default"
      type="button"
    >
      Open
    </button>
  </div>
</div>
`;

exports[`Drawer Demo HeaderUsageExample wroks fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        <div>
          自定义头部
        </div>
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is the body of drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <button
          class="t-button t-size-m t-button--variant-base t-button--theme-default"
          type="button"
        >
          确定
        </button>
         
        <button
          class="t-button t-size-m t-button--variant-outline t-button--theme-default"
          type="button"
        >
          取消
        </button>
      </div>
    </div>
  </div>
   
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        This is the title.
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is the body of drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            confrim
          </button>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            cancel
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        I am the title
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is the body of drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            cancel
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default t-is-disabled"
            disabled="disabled"
            type="button"
          >
            OK
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-default"
    type="button"
  >
    Open(插槽定义)
  </button>
  
    
  
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-default"
    type="button"
  >
    Open(属性定义)
  </button>
  
    
  
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-default"
    type="button"
  >
    Open(单独设置确认/取消按钮)
  </button>
</div>
`;

exports[`Drawer Demo ModeUsageExample wroks fine 1`] = `
<div>
  <!---->
   
  <div
    class="tdesign-radio-button"
  >
    
    抽屉弹出方向：
    
    <div
      class="t-radio-group t-radio-group-outline t-radio-group-medium"
    >
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="left"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          left
        </span>
      </label>
       
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="right"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          right
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="top"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          top
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="bottom"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          bottom
        </span>
      </label>
    </div>
  </div>
   
  <br />
   
  <div
    class="tdesign-radio-button"
  >
    
    抽屉弹出模式：
    
    <div
      class="t-radio-group t-radio-group-outline t-radio-group-medium"
    >
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="overlay"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          overlay
        </span>
      </label>
       
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="push"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          push
        </span>
      </label>
    </div>
  </div>
   
  <br />
  <br />
   
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-primary"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`Drawer Demo NoMaskUsageExample wroks fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer-right t-drawer-no-mask v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        标题名称
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is a drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            取消
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            确认
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-primary"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`Drawer Demo PlacementUsageExample wroks fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        标题名称
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is a drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            取消
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            确认
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="tdesign-radio-button"
  >
    <div
      class="t-radio-group t-radio-group-outline t-radio-group-medium"
    >
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="left"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          left
        </span>
      </label>
       
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="right"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          right
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="top"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          top
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="bottom"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          bottom
        </span>
      </label>
    </div>
     
    <button
      class="t-button t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      Open
    </button>
  </div>
</div>
`;

exports[`Drawer Demo SizeUsageExample wroks fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        标题名称
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is a drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            取消
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            确认
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="tdesign-radio-button"
  >
    <div
      class="t-radio-group t-radio-group-outline t-radio-group-medium"
    >
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="left"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          left
        </span>
      </label>
       
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="right"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          right
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="top"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          top
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="bottom"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          bottom
        </span>
      </label>
    </div>
  </div>
   
  <div
    class="tdesign-radio-button"
  >
    <div
      class="btn-top-margin t-radio-group t-radio-group-outline t-radio-group-medium"
    >
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="small"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          small(300px)
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="medium"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          medium(500px)
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="large"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          large(760px)
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="200"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          200
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="400px"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          400px
        </span>
      </label>
       
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="50%"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          50%
        </span>
      </label>
    </div>
  </div>
   
  <button
    class="btn-top-margin t-button t-size-m t-button--variant-base t-button--theme-primary"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`Drawer Demo destroy wroks fine 1`] = `
<div>
  <!---->
   
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-default"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`Drawer Demo operation wroks fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        标题名称
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <div>
          <span>
            Label A
          </span>
           
          <div
            class="t-input t-size-m"
          >
            <input
              class="t-input__inner"
              type="text"
            />
          </div>
        </div>
         
        <div>
          <span>
            Label B
          </span>
           
          <div
            class="t-input t-size-m"
          >
            <input
              class="t-input__inner"
              type="text"
            />
          </div>
        </div>
         
        <div>
          <span>
            Label C
          </span>
           
          <div
            class="t-input t-size-m"
          >
            <input
              class="t-input__inner"
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            取消
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            确认
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-primary"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`Drawer base demo works fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer-right v-transfer-dom"
    style="z-index: 1500;"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper-right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        Drawer
      </div>
      <div
        class="t-drawer__close-btn"
      />
      <div
        class="t-drawer__body"
      >
        <p>
          This is a controlled drawer
        </p>
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            取消
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            确认
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-primary"
    type="button"
  >
    Open a Drawer
  </button>
</div>
`;
